examples/community/unclip_text_interpolation.py

```diff
@@ -0,0 +1,673 @@
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether or not to return a [`~pipelines.ImagePipelineOutput`] instead of a plain tuple.
+        """
+
+        if enable_sequential_cpu_offload:
+            self.enable_sequential_cpu_offload(gpu_id=gpu_id)
+
+        device = self._execution_device
+
+        # Turn the prompts into embeddings.
+        inputs = self.tokenizer(
+            [start_prompt, end_prompt],
+            padding="max_length",
+            truncation=True,
+            max_length=self.tokenizer.model_max_length,
+            return_tensors="pt",
+        )
+        inputs.to(device)
+        text_model_output = self.text_encoder(**inputs)
+
+        attention_mask = torch.max(inputs.attention_mask[0], inputs.attention_mask[1])
+        attention_mask = attention_mask.unsqueeze(0).to(device)
+
+        start_text_embeds = text_model_output.text_embeds[0]
+        end_text_embeds = text_model_output.text_embeds[1]
+
+        start_last_hidden_state = text_model_output.last_hidden_state[0]
+        end_last_hidden_state = text_model_output.last_hidden_state[1]
+
+        # Interpolate from the start to end prompt using slerp and add the generated images to an image output pipeline
+        out = []
+        for interp_val in np.linspace(0, 1, steps):
+            # Use the start and end prompts for 0 and 1 values as slerp results are subjectively worse than slerp results for the same.
+            if interp_val == 0:
+                text_embeds = start_text_embeds
+                last_hidden_state = start_last_hidden_state
+            elif interp_val == 1:
+                text_embeds = end_text_embeds
+                last_hidden_state = end_last_hidden_state
+            else:
+                text_embeds = UnCLIPTextInterpolationPipeline.slerp(interp_val, start_text_embeds, end_text_embeds)
+                last_hidden_state = UnCLIPTextInterpolationPipeline.slerp(
+                    interp_val, start_last_hidden_state, end_last_hidden_state
+                )
+
+            text_model_output.text_embeds = text_embeds.unsqueeze(0).to(device)
+            text_model_output.last_hidden_state = last_hidden_state.unsqueeze(0).to(device)
+
+            res = self._generate(
+                text_model_output=text_model_output,
+                text_attention_mask=attention_mask,
+                generator=generator,
+                prior_num_inference_steps=prior_num_inference_steps,
+                decoder_num_inference_steps=decoder_num_inference_steps,
+                super_res_num_inference_steps=super_res_num_inference_steps,
+                prior_guidance_scale=prior_guidance_scale,
+                decoder_guidance_scale=decoder_guidance_scale,
+                output_type=output_type,
+                return_dict=return_dict,
+            )
+
```

Comment:
I think ideally we should batch the embeddings instead of effectively running the pipeline in a loop

Reply:
Sure. Will batch the pipeline run.